configure.py

```diff
@@ -1373,6 +1381,11 @@ def write_config(data, name):
     locs.add('root')  # must have root
     o['variables']['icu_locales'] = ','.join(str(loc) for loc in locs)
     # We will check a bit later if we can use the canned deps/icu-small
+    o['variables']['icu_has_default_data_dir'] = b(False)
+    if bool(options.with_icu_default_data_dir):
+      o['variables']['icu_has_default_data_dir'] = b(True)
+      o['variables']['icu_default_data_dir'] = options.with_icu_default_data_dir      # Needs to be set but empty for node.gypi to process
```

Comment:
Long line. FWIW, I'd just set it like this:
```python
o['variables']['icu_default_data_dir'] = options.with_icu_default_data_dir or ''
```
(e.g., no if stmt, no `icu_has_default_data_dir`) and then just check that `icu_default_data_dir!=""` in the gyp files.

Reply:
I actually tried exactly this first, but for some reason the `icu_default_data_dir!=''` was always failing the comparison. I'm not sure why. I'll try again.

Reply:
Seems to be working in that construct, so :man_shrugging:

Reply:
Nope, I see what it was now. It appears that gyp treats variables suffixed with `_dir` specially and it was throwing:

```
Traceback (most recent call last):
  File "./configure", line 28, in <module>
    import configure
  File "/home/sgallagh/workspace/node/configure.py", line 1729, in <module>
    run_gyp(gyp_args)
  File "tools/gyp_node.py", line 54, in run_gyp
    rc = gyp.main(args)
  File "tools/gyp/pylib/gyp/__init__.py", line 547, in main
    return gyp_main(args)
  File "tools/gyp/pylib/gyp/__init__.py", line 523, in gyp_main
    options.duplicate_basename_check)
  File "tools/gyp/pylib/gyp/__init__.py", line 139, in Load
    params['parallel'], params['root_targets'])
  File "tools/gyp/pylib/gyp/input.py", line 2785, in Load
    variables, includes, depth, check, True)
  File "tools/gyp/pylib/gyp/input.py", line 383, in LoadTargetBuildFile
    includes, True, check)
  File "tools/gyp/pylib/gyp/input.py", line 255, in LoadOneBuildFile
    aux_data, includes, check)
  File "tools/gyp/pylib/gyp/input.py", line 293, in LoadBuildFileIncludesIntoDict
    subdict_path, include)
  File "tools/gyp/pylib/gyp/input.py", line 2151, in MergeDicts
    MergeDicts(to[k], v, to_file, fro_file)
  File "tools/gyp/pylib/gyp/input.py", line 2144, in MergeDicts
    to[k] = MakePathRelative(to_file, fro_file, v)
  File "tools/gyp/pylib/gyp/input.py", line 2043, in MakePathRelative
    if item[-1] == '/':
IndexError: string index out of range while reading includes of /home/sgallagh/workspace/node/node.gyp while trying to load /home/sgallagh/workspace/node/node.gyp
```

After I drop the `_dir` suffix (leaving `icu_default_data`), it works properly.